objgraph.py

```diff
@@ -857,26 +868,38 @@ def _safe_repr(obj):
         return '(unrepresentable)'
 
 
+def _name_or_repr(value):
+    try:
+        result = value.__name__
+    except AttributeError:
+        result = repr(value)[:40]
+
+    if _isinstance(result, basestring):
+        return result
+    else:
+        return repr(value)[:40]
+
+
 def _short_repr(obj):
-    if isinstance(obj, (type, types.ModuleType, types.BuiltinMethodType,
-                        types.BuiltinFunctionType)):
-        return obj.__name__
-    if isinstance(obj, types.MethodType):
+    if _isinstance(obj, (type, types.ModuleType, types.BuiltinMethodType,
+                         types.BuiltinFunctionType)):
+        return _name_or_repr(obj)
+    if _isinstance(obj, types.MethodType):
         try:
-            if obj.__self__ is not None:
-                return obj.__func__.__name__ + ' (bound)'
-            else:
-                return obj.__func__.__name__
+            func = obj.__func__
+            instance = obj.__self__
         except AttributeError:  # pragma: nocover
             # Python < 2.6 compatibility
```

Comment:
You can  simplify this by dropping the branch that attempts to support Python 2.5 and older.